crates/bootstrap_srv/src/store.rs

```diff
@@ -0,0 +1,191 @@
+//! This is a virtual-memory inspired tempfile store.
+//!
+//! ### Rationale
+//!
+//! - We don't need to persist anything in the store beyond a single
+//!   process invocation, because the infos are going to expire after
+//!   a matter of minutes anyways, and peers will continue to re-publish them.
+//! - We would like a server to be able to store more spaces and infos
+//!   than would reasonably fit in RAM.
+//!
+//! ### Implementation
+//!
+//! - Have a pool of tempfiles that can grow to match the worker thread
```

Comment:
I'm confused by this line -- it sounds like the max number of tempfiles is the worker_thread_count?

Reply:
Yes exactly. Since we're not using async code, the items from the pool can only be checked out by an active thread, and then they are returned before that thread can check out another. So the worker thread count is the *implicit* max count of active temp files that can be checked out from the pool.

There is a nuance here, however: when the tempfiles roll over from size, they are not re-added to the pool. So the pool will never grow beyond that, but while there are still read refs held for these other items, they will remain in memory.